labs/oas-apigee-mock/lab.md

```diff
@@ -0,0 +1,181 @@
+the process shown below. An Open API specification and test suite have already
+been provided for the lab. You can use these as a reference
+to leverge this same process in your own API development work.
+
+![Overview](assets/overview.png)
+
+We assume the basic knowledge of the Apigee platform and you will get the most
+from this lab if you already have this knowledge.
+
+Ideally you will have completed the Apigee courses on
+[Design](https://www.coursera.org/learn/api-design-apigee-gcp),
+[Development](https://www.coursera.org/learn/api-development-apigee-gcp) and
+[Security](https://www.coursera.org/learn/api-security-apigee-gcp) Courses.
+These are available through Coursera, Pluralsight or Qwiklabs.
+
+Alternatively, completing the Apigee
+[API Jam](https://github.com/apigee/apijam) will cover the same topics in
+less depth.
+
+Lets get started!
+
+## Prerequisites
+
+Duration: 0:30:00
+
+An Apigee X organisation configured for external exposure is required to
+complete this lab. See
+[here](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+for details on provisioning an evaluation organisation.
+
+### Tools
+
+You will need to have the following dependencies configired in you local
+enviornment in order to complete the lab tasks:
+
+- Bash (Unix shell)
+- [NodeJS](https://nodejs.org/en/) LTS version or above.
+- [gcloud](https://cloud.google.com/sdk/docs/install)
+- [git](https://git-scm.com/)
+- [apigee-sackmesser](https://github.com/apigee/devrel/tree/main/tools/apigee-sackmesser)
+- [apigeecli](https://github.com/srinandan/apigeecli)
+- [jq](https://stedolan.github.io/jq/)
+
+### Environment Variables
+
+Commands used during the lab will require the following environment variables
+to be configured.
+
+``` bash
+export TOKEN=$(gcloud auth application-default print-access-token)
+export APIGEE_ENV=<apigee-environment-name>
+export APIGEE_ORG=<gcp-project-name>
+export RUNTIME_HOST_ALIAS=<your-runtime-host-alias>
+```
+
+### oas-apigee-mock
+
+Clone the Apigee DevRel repo and install the dependencies for the
+oas-apigee-mock tool.
+
+``` bash
+git clone https://github.com/apigee/devrel.git
+cd devrel/tools/oas-apigee-mock/
+npm install
+```
+
+## Generate, deploy and test an Apigee Proxy
+
+Duration: 0:30:00
+
+### Generate an Apigee Proxy from an Open API Specification
+
+Use the `orders-apikey-header.yaml` Open API specification included
+in the `devrel/tools/oas-apigee-mock/test/oas` folder to generate a proxy bundle.
+
+``` bash
+node bin/oas-apigee-mock generateApi web-orders-proxy-v1 \
+  -s test/oas/orders-apikey-header.yaml \
+  -o
+```
+
+### Update the apickli configuration to use your Host Alias
+
+Update the `before(function ()` block in
+`test/features/step_definitions/init.js` with your organisation's hostname.
+This will be your `RUNTIME_HOST_ALIAS` if you followed the
+[Apigee X Trial Provisioning](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+script.
```

Comment:
Not entirely clear what `This` is referencing here. Maybe you can put in the curl command to list the envgroups and jq to extract the hostnames instead?